mandelbrot-set-python/mandelbrot_03.py

```diff
@@ -0,0 +1,27 @@
+# mandelbrot_03.py
+
+from dataclasses import dataclass
+from math import log
+
+
+@dataclass
+class MandelbrotSet:
+    max_iterations: int
+    escape_radius: float = 2.0
+
+    def __contains__(self, c: complex) -> bool:
+        return self.stability(c) == 1
+
+    def stability(self, c: complex, smooth=False, clamp=True) -> float:
+        value = self.escape_count(c, smooth) / self.max_iterations
+        return max(0.0, min(value, 1.0)) if clamp else value
+
+    def escape_count(self, c: complex, smooth=False) -> int | float:
```

Comment:
Out of the box, the `int | float` syntax means that this only runs on Python 3.10+. However, we can add a `from __future__ import annotations` on the top of this file to make the syntax supported on older versions, and this module work on Python 3.7+

Reply:
Nice trick! I haven't been using `from __future__ import ...` and almost forgot about its existence. Thanks :wink: